### PR TITLE
support in transformer for assignments

### PR DESCRIPTION
**Pull request by tomblind**
Everything works (tests are passing), but this could probably use some cleanup/refactoring.

Note, I had to do some split code paths to deal with statements vs expressions.

I noticed in the TS source, operators like +=, -=, etc... are referred to as 'compound assignment operators" so I'm sticking with that terminology for those.

### CODE REVIEW

**Comment by tomblind** [comment]
Looks like I have tests to fix...

File: src/TSHelper.ts [hunk 1]
```diff
@@ -220,12 +222,50 @@ export class TSHelper {
         return [false, null];
     }
 
+    public static isNonCompoundPrefixUnaryOperator(node: ts.PrefixUnaryExpression): boolean {
```
Comment (Perryvw) [inline src/TSHelper.ts]: Would it not be nicer to have the opposite of this function and just use a negation where you need this? Depends on what the most common case is I guess.

File: src/TSHelper.ts [hunk 2]
```diff
@@ -220,12 +222,50 @@ export class TSHelper {
...
+        }
+    }
+
     public static isExpressionStatement(node: ts.Expression): boolean {
         return node.parent === undefined
             || ts.isExpressionStatement(node.parent)
             || ts.isForStatement(node.parent);
     }
 
+    public static createLuaConcatExpression(left: ts.Expression, right: ts.Expression): LuaConcatExpression {
```
Comment (Perryvw) [inline src/TSHelper.ts]: I think @lolleko made a TransformHelper.ts for functions like these.

Comment (tomblind) [reply src/TSHelper.ts]: Totally missed it! I'll move these over.

**Pull request merged**